x/feegrant/msgs.go

```diff
@@ -41,21 +41,7 @@ func NewMsgGrantAllowance(feeAllowance FeeAllowanceI, granter, grantee sdk.AccAd
 
 // ValidateBasic implements the sdk.Msg interface.
 func (msg MsgGrantAllowance) ValidateBasic() error {
-	if _, err := sdk.AccAddressFromBech32(msg.Granter); err != nil {
-		return sdkerrors.ErrInvalidAddress.Wrapf("invalid granter address: %s", err)
-	}
-	if _, err := sdk.AccAddressFromBech32(msg.Grantee); err != nil {
-		return sdkerrors.ErrInvalidAddress.Wrapf("invalid grantee address: %s", err)
-	}
-	if msg.Grantee == msg.Granter {
-		return errorsmod.Wrap(sdkerrors.ErrInvalidAddress, "cannot self-grant fee authorization")
-	}
-	allowance, err := msg.GetFeeAllowanceI()
-	if err != nil {
-		return err
-	}
-
-	return allowance.ValidateBasic()
+	return nil
```

Comment:
Cannot we compare the bytes of the two addresses and keep the `msg.Grantee == msg.Granter` check? This seems still valid to me.
I see you've done that in the msgServer now, maybe I am missing the point, but why?

Reply:
this was explained in rfc001, we would move towards deprecating valdiate basic

Reply:
Thank you, I knew I was missing something :)